sky/backends/backend_utils.py

```diff
@@ -758,7 +759,10 @@ def write_cluster_config(
     yaml_path = _get_yaml_path_from_cluster_name(cluster_name)
 
     # Use a tmp file path to avoid incomplete YAML file being re-used in the future.
-    tmp_yaml_path = yaml_path + '.tmp'
+    if isinstance(cloud, clouds.Local):
+        tmp_yaml_path = yaml_path
+    else:
+        tmp_yaml_path = yaml_path + '.tmp'
```

Comment:
Sorry could you explain why is this if-else needed?

Reply:
Yes, in the generated local YAML, the file mounts field has `~/.sky/sky_yaml: ~/.sky/generated/local-cluster.yaml.tmp`. The `.tmp` file does not exist by then, so there is a `ray up error`. This is just preventing the `.tmp` file from being created, since the logic for local ray yaml is diff.

Reply:
Ah thanks. now I see why we failed to capture this. In `fill_template` we fill in some additional template fields 
https://github.com/skypilot-org/skypilot/blob/4abd501fdee3a6f7049d8229ae4ddee3175a7430/sky/backends/backend_utils.py#L162-L164

Maybe we should consider moving the two to the dictionary we pass to `fill_template`?
https://github.com/skypilot-org/skypilot/blob/4abd501fdee3a6f7049d8229ae4ddee3175a7430/sky/backends/backend_utils.py#L766-L799

So we just need to pass
```
sky_ray_yaml_local_path: yaml_path
sky_ray_yaml_remote_path: SKY_RAY_YAML_REMOTE_PATH
```

and if-else is not needed. I think it'd be less error prone and consistent?

Reply:
Done, thanks!